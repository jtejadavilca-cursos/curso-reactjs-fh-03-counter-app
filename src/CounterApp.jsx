import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState( value );

    const handleAdd = () => setCounter( counter + 1 );
    const handleSubs = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

  return (
    <>
        <h1>CounterApp</h1>
        <h1> { counter } </h1>

        <button onClick={ handleAdd }> +1 </button>
        <button onClick={ handleSubs }> -1 </button>
        <button onClick={ handleReset }> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: -1
}