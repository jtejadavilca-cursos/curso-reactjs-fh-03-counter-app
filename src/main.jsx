import React from 'react';
import ReactDOM from 'react-dom/client';

import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';
//import { HelloWorldApp } from './HelloWorldApp';

import './css/styles.css'

ReactDOM.createRoot( document.getElementById( 'root' ) )
.render(
    <React.StrictMode>
        <CounterApp value={10} />

        {/* <FirstApp title="Hola, soy Goku" /> */}
    </React.StrictMode>
);