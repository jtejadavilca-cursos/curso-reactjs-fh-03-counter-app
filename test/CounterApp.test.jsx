// Tarea
// Pruebas en el <CounterApp />

import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

// test: debe de hacer match con el snapshot
// test: debe de mostrar el valor inicial de 100 <CounterApp value={100} />
describe('Pruebas en el <CounterApp />', () => {

    test('debe de hacer match con el snapshot', () => {
        const value = 10;
        const { container } = render(<CounterApp value={value} />);

        expect( container ).toMatchSnapshot();
        expect( screen.getByText(value) ).toBeTruthy();
    });

    test('debe de mostrar el valor inicial de 100 <CounterApp value={100}', () => {
        const value = 100;
        const { container } = render(<CounterApp value={value} />);

        expect( screen.getAllByRole('heading', {level: 1})[1].innerHTML.trim() ).toContain( `${value}` );
        expect( screen.getAllByText(`${value}`).length ).toBe(1);
    });

    test('debe de incrementar con el botón +1', () => {
        const value = 10;
        render(<CounterApp value={value} />);

        fireEvent.click( screen.getByText('+1') );

        expect( screen.getByText('11') ).toBeTruthy();
    });

    test('debe de derementar con el botón -1', () => {
        const value = 10;
        render(<CounterApp value={value} />);

        fireEvent.click( screen.getByText('-1') );

        expect( screen.getByText('9') ).toBeTruthy();
    });

    test('debe de reseteaer con el botón Reset', () => {
        const value = 10;
        render(<CounterApp value={value} />);


        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('12') ).toBeTruthy();
        fireEvent.click( screen.getByText('Reset') );

        expect( screen.getByText('10') ).toBeTruthy();
    });
});