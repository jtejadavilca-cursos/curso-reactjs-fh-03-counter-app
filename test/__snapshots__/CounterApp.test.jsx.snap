// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h1>
     
    10
     
  </h1>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
